internal/provider/data_source_http.go

```diff
@@ -317,6 +322,12 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 		request.Header.Set(name, header)
 	}
 
+	hostHeader := request.Header.Get("Host")
+	if hostHeader != "" && model.AllowHostOverride.ValueBool() {
+		request.Host = hostHeader
+		request.URL.Host = hostHeader
```

Comment:
```suggestion
```

I don't believe we want to override the actual URL being connected to when this is present, just the `Host` header itself.

See: https://github.com/golang/go/blob/7321aa91c6c7f4c7faa59819a2013971712eba84/src/net/http/request.go#L125-L129